qiskit/circuit/quantumcircuit.py

```diff
@@ -1480,13 +1480,17 @@ def decompose(
         gates_to_decompose: Optional[
             Union[Type[Gate], Sequence[Type[Gate]], Sequence[str], str]
         ] = None,
+        reps: int = 1,
     ) -> "QuantumCircuit":
         """Call a decomposition pass on this circuit,
         to decompose one level (shallow decompose).
 
         Args:
             gates_to_decompose (str or list(str)): optional subset of gates to decompose.
                 Defaults to all gates in circuit.
+            reps (int): Optional number of times the circuit should be decomposed.
+                For instance, ``reps=2`` equals calling ``circuit.decompose().decompose()``.
+                can decompose specific gates specific time
```

Comment:
Just to explain myself from before: this text (which I agree with) is why I didn't want to try and do "magic" things with the `gates_to_decompose` argument.  If we did that, `circuit.decompose(gates, reps=2)` wouldn't be the same as `circuit.decompose(gates).decompose(gates)`.

(I also didn't like the stateful effects of the `run` from a somewhat inherently stateless pass, but that's beside the point now.)